queue/QueueConsumer.js

```diff
@@ -24,13 +52,57 @@ module.exports = class QueueConsumer extends BaseQueue {
     if (!this._channel) {
       throw new Error('Channel not initialized')
     }
+
+    const messageContent = message.content
+
+    const maxMessageRetries = this._maxMessageRetries
+    const deadLetterQueueName = this._deadLetterQueueName
+
+    const { headers } = message.properties
+    const deadLetterHeader = headers[DEAD_LETTER_HEADER]
+
+    let rejectedRecord
+    if (deadLetterHeader) {
+      // find 'rejected' record
+      // TODO: handle other reasons in the future
+      // ex. dead letter messages from TTL expires (not implemented yet)
+      for (const record of deadLetterHeader) {
+        if (record.reason === 'rejected') {
+          rejectedRecord = record
+        }
+      }
+    }
+
+    if (rejectedRecord) {
+      // get the number of times this message has been attempted
+      const attemptCount = rejectedRecord.count
+
+      if (attemptCount < maxMessageRetries) {
```

Comment:
I don't think `maxMessageSendAttempts` is right. I do agree that the current name is somewhat off. Maybe something like `maxMessageRejections` is better since this is used when rejecting a message.